### PR TITLE
Add Manual Match Functionality and Tests

### PR DESCRIPTION
**Pull request by craig-riecke**
Implements #94.

### CODE REVIEW

**Comment by craig-riecke** [comment]
Would someone take a look at this when you have a chance?  Thanks!

File: app/models/recipient.rb [hunk 1]
```diff
@@ -27,4 +27,19 @@ class Recipient < ActiveRecord::Base
...
+
+  # Race and size are only listed where relevant, for dolls and clothes respectively
+  def other_details
+    [ race.blank? ? nil : "Race: #{race}",
+      size.blank? ? nil : "Size: #{size}"
+    ].compact.join(", ")
+  end
+
+  def matched?
+    ! membership_id.nil?
```
Comment (atzorvas) [inline app/models/recipient.rb]: is there any reason to not use .present?

File: app/controllers/memberships_controller.rb [hunk 2]
```diff
@@ -16,14 +16,21 @@ def create
     # Create the user first, but don't duplicate!  Generate a phony password - this will
     # be overwritten on the user's first login
     user = User.find_by(email: user_params[:email])
-    create_user_for_membership unless user
+    user = create_user_for_membership unless user
```
Comment (atzorvas) [inline app/controllers/memberships_controller.rb]: Personally, I would prefer: 

`user ||= create_user_for_membership` 
or

```
user = User.find_by(email: user_params[:email]) || 
     create_user_for_membership
```

File: app/controllers/recipients_controller.rb [hunk 3]
```diff
@@ -1,12 +1,29 @@
...
+    # Destination depends on where match came from: org admin, or user
+    if params[:organization_id]
+      @org = Organization.friendly.find(params[:organization_id])
+      redirect_to organization_recipients_path(@org)
+    else
+      flash[:message] = "Thank you for helping #{recipient.first_name}!"
+      redirect_to user_show_path
+    end
   end
+
```
Comment (atzorvas) [inline app/controllers/recipients_controller.rb]: prefer to clear this empty line

**Comment by atzorvas** [comment]
@craig-riecke :+1: for the functionality

if you agree with the comments you can continue with doing any cleanups 
feel free to merge this nevertheless :tada:

**Comment by craig-riecke** [comment]
@atzorvas Your comments made sense, so I cleaned up those things, and also some things Rubocop found.  Thanks!  Merging...

**Pull request merged**